trlx/pipeline/offline_pipeline.py

```diff
@@ -125,7 +134,15 @@ def __len__(self) -> int:
         return len(self.prompts)
 
     def create_loader(self, batch_size: int, shuffle=False) -> DataLoader:
-        collate_fn = DataCollatorWithPadding(self.tokenizer) if self.tokenizer else torch.vstack
+        def collate_fn(xs):
+            out = self.tokenizer.pad([{"input_ids": x["input_ids"]} for x in xs], return_tensors="pt")
```

Comment:
Is it safe to assume `self.tokenizer` will always be available? Not sure if the previous check `if self.tokenizer else torch.vstack` is just stale code?

Reply:
It was a stale code and the single example which didn't use tokenizer at the time (ilql_randomwalks) was also converted to use tokenizer to drop these checks, out of which it is the last, for consistency; right now tokenizer is required in almost every other piece of code, so I'm not sure if that feature is missed.